modules/geolocation/geolocation.py

```diff
@@ -129,3 +140,210 @@ def calculate_pixel_to_geo_mapping(self):
+        c = self.__calculate_c_vector(compoundRotationMatrix)   # use default camera direction
+        u = self.__calculate_u_vector(compoundRotationMatrix)   # use default u vector direction
+        v = self.__calculate_v_vector(c, u)
+
+        return (o, c, u, v)
+
+    def __calculate_o_vector(self, latitude: int, longitude: int, altitude: int, gpsCameraOffset:np.ndarray, cameraRotation: np.ndarray, planeRotation: np.ndarray, origin: np.ndarray) -> np.ndarray:
+        """
+        Returns a numpy array that contains the components of the o vector
+
+        Parameters
+        ----------
+        latitude: int
+            The GPS latitude of the plane
+        longitude: int
+            The GPS longitude of the plane
+        altitude: int
+            The GPS altitude of the plane
+        gpsCameraOffset: numpy array
+            The offset of the GPS to the camera on the plane in plane space
+        cameraRotation: numpy array
+            Array containing rotation matrix for camera rotation
+        planeRotation: numpy array
+            Array containing rotation matrix for plane rotation
+        origin:numpy array
+            Array containing the origin of the world space
+
+        Returns
+        -------
+        oVector: numpy array
+            Array containing components of the o vector
+        """
+        # get GPS module coordinates
+        gpsModule = np.zeros(3)
+        gpsModule[0] = origin[0] + latitude
+        gpsModule[1] = origin[1] + longitude
+        gpsModule[2] = origin[2] + altitude
+
+        # transform gps-to-camera offset from plane space to world space by applying inverse
+        # rotation matrix using (AB)^-1 = B^-1 A^-1
+        rotationMatrix = np.matmul(planeRotation, cameraRotation)
+        transposedMatrix = np.transpose(rotationMatrix)
+        rotatedOffset = transposedMatrix.dot(gpsCameraOffset)
+
+        # get camera coordinates in world space
+        oVector = np.add(rotatedOffset, gpsModule)
+
+        return oVector
+
+    def __calculate_c_vector(self, compoundRotationMatrix: np.ndarray, cVectorCameraSpace=np.array([1, 0, 0])) -> np.ndarray:
+        """
+        Returns a numpy array that contains the components of the c vector
+
+        Parameters
+        ----------
+        compoundRotationMatrix: numpy array
+            Array containing rotation matrix for camera and plane rotations
+        cVectorCameraSpace: numpy array
+            Array containing cVector in camera space (default value assumes cVector points along the roll axis)
+
+        Returns
+        -------
+        cVector:
+            Array containing components of the c vector
+        """
+        # apply rotation matrix to camera direction
+        # note: this assumes that camera euler angles are w.r.t. plane space
+        cVector = compoundRotationMatrix.dot(cVectorCameraSpace)
+
+        # normalize since magnitude doesn't matter
+        norm = np.linalg.norm(cVector)
+        if(norm != 0) :
```

Comment:
Formatting nitpick: Space between if and open parantheses, no space between closing parantheses and colon.